src/main/java/org/opensearch/searchrelevance/judgments/ImportJudgmentsProcessor.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.searchrelevance.judgments;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.searchrelevance.model.JudgmentType;
+import org.opensearch.transport.client.Client;
+
+public class ImportJudgmentsProcessor implements BaseJudgmentsProcessor {
+    private static final Logger LOGGER = LogManager.getLogger(ImportJudgmentsProcessor.class);
+    private final Client client;
+
+    @Inject
+    public ImportJudgmentsProcessor(Client client) {
+        this.client = client;
+    }
+
+    @Override
+    public JudgmentType getJudgmentType() {
+        return JudgmentType.IMPORT_JUDGMENT;
+    }
+
+    @Override
+    public void generateJudgmentScore(Map<String, Object> metadata, ActionListener<Map<String, Map<String, String>>> listener) {
+
+        Map<String, Object> sourceJudgementScores = (Map<String, Object>) metadata.get("judgmentScores");
+        metadata.remove("judgmentScores");
+
+        // Create the result map in the expected format
+        Map<String, Map<String, String>> formattedScores = new HashMap<>();
+
+        // Process each query
+        for (Map.Entry<String, Object> queryEntry : sourceJudgementScores.entrySet()) {
+            String queryText = queryEntry.getKey();
+            Object scoreData = queryEntry.getValue();
+
+            // Create a map for this query's document scores
+            Map<String, String> docScores = new HashMap<>();
+
+            // Handle the case when scoreData is a List (from JSON array)
+            if (scoreData instanceof List) {
```

Comment:
in case scoreData is not a list, we do nothing. Is this expected and can we at least put to log some message?

Reply:
honestly, i think we don't need special handling, casue if it's not a list, well, then the whole api structure is wrong and it should blow up!

Reply:
I think we need to act somehow if this is not a List. It can help to troubleshoot problems later, for instance if user uploaded data in unsupported format. If you want to signal that back to user immediately then exception message can be informative so they know where the problem is. From my point of view it's better to fail fast rather then silently accept wrong input and then went into corrupted system state later

Reply:
done!